tests/MemoryLeakWarningTest.cpp

```diff
@@ -385,8 +440,49 @@ TEST(MemoryLeakWarningWarningThreadSafe, turnOnThreadSafeNewDeleteOverloads)
     MemoryLeakWarningPlugin::turnOnNewDeleteOverloads();
 }
 
+#ifdef __clang__
+
+IGNORE_TEST(MemoryLeakWarningThreadSafe, turnOnThreadSafeNewDeleteOverloads)
+{
+    /*  Clang misbehaves with -O2 - it will not overload operator new or
+     *  operator new[] no matter what. Therefore, this test is must be ignored.
+     */
+}
```

Comment:
I actually think that I once made a bug report to clang about this. Don't think they ever fixed it, neither remember seeing it closed.